### ISSUE TITLE
Object detection training on a coco label with custom dataset

### ISSUE THREAD

**Issue by Kbkartik** [facebookresearch/detectron2#662]
Hi, I have coco-format custom dataset containing train/validation sets with 'person' label only. I have registered the dataset successfully using register_coco_instances. While training & validation, will detectron2 create a new 'person' label or map it with the already existing 'person' label in the pre-trained models? I am looking for the latter. Also, is there a way to verify the latter?

Thanks!

**Comment by ppwwyyxx** [comment]
If your COCO format is correct then labels will be handled automatically. When the training dataloader is created it will print some statistics of training data & labels and you can use it to verify.

**Comment by Kbkartik** [comment]
I see. I trained the model on an awfully small training dataset (2 images consisting of 3 bounding boxes) just to check if it is mapping the label. However, I get the following output:

![new_label](https://user-images.githubusercontent.com/30231581/72149369-fc719780-33c8-11ea-8a87-9024d2f5629d.png)

This means the label has been mapped with the pre-trained model right?

**Comment by ppwwyyxx** [comment]
Sorry for the confusion. The labels do not get mapped to pre-trained models - they will be handled as normal new labels. And as the log says those weights are skipped. In fact a pre-trained model don't even store class names.

If you want to achieve such things you can modify the weights of the pre-trained model to match the new dataset.

**Comment by Kbkartik** [comment]
When you say the weights are skipped, are you talking about the fully connected layer at the end of the model?

Also, by modifying the weights of the pre-trained model, do you mean I further train the pre-trained model on my dataset and make predictions based on my dataset's label ('person' in this case). This approach would be the same as using a pre-trained MSCOCO model from model except for the enhancement of features for a specific label. Is my understanding correct?

**Comment by ppwwyyxx** [comment]
layers that have different sizes (e.g. classes) will be thrown out.

by modifying I mean you can edit the weights to have the same size, and align "person" in one model to "person" in another.

**Comment by Kbkartik** [comment]
So I am a bit unclear on "modifying/editing the weights"? Could you please elaborate on the same?

Thanks!

**Comment by ppwwyyxx** [comment]
You can use `torch.load/save` to read/write the weights.

**Comment by Kbkartik** [comment]
Ah, I see. Also, in plain_train_net.py, I see this [cfg.freeze](https://github.com/facebookresearch/detectron2/blob/master/tools/plain_train_net.py#L192).

Here, should we be unfreezing the parameters while training?

**Comment by ppwwyyxx** [comment]
It's unrelated to parameters. 

By default only the first few layers are frozen and can be controlled by `MODEL.BACKBONE.FREEZE_AT=0`.

**Comment by Kbkartik** [comment]
Just curious. What does that line correspond to anyway?
